### ISSUE TITLE
V10 EntryLink types aren't correct without any chain modifiers

### ISSUE THREAD

**Issue by gigaSproule** [contentful/contentful.js#1897]
## Expected Behavior

When using the `contentful` client in TypeScript, with a valid type definition containing an `EntryLink` and no chain modifiers are used as part of the call to get an entry, then the type should be aware that the fields of the linked entry will be returned.

## Actual Behavior

The type definition of the returned EntryLink only contains `sys`.

## Possible Solution

I'm guessing that the issue is to do with the `ResolvedEntryLink` type definition.

## Steps to Reproduce

Using `contentful` v10.1.3, the following code with give a compilation error.

```
import { createClient, EntryFieldTypes, EntrySkeletonType } from "contentful";

const doSomething = async () => {
  const client = createClient({
    space: "****",
    environment: "****",
    accessToken: "****"
  });

  const entry = await client.getEntry<
    EntrySkeletonType<
      {
        linkedEntry: EntryFieldTypes.EntryLink<{
          fields: {
            myField: EntryFieldTypes.Text;
          };
          contentTypeId: "MyLinkedContentType";
        }>;
      },
      "MyContentType"
    >
  >("id");
  const a = entry.fields.linkedEntry.sys;
  const b = entry.fields.linkedEntry.fields;
};
```

## Context

I am trying to accurately type our data model from Contentful, so it can be easily used in our code.

## Environment

-   **Language Version**: v18.16.0
-   **Package Manager Version**: yarn 3.5.0
-   **Browser Version**: N/A
-   **Operating System**: Darwin MacBook-Pro.localdomain 22.4.0 Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:28 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6000 arm64
-   **Package Version**: 10.1.3
-   **Which API are you using?**: Delivery

**Comment by veu** [comment]
The default mode without chain modifiers is "try to fetch linked entries, replace links with fetched entries, and leave links in place where entries could not be fetched". Because we don’t know on the type level whether entries can be fetched, you’ll get a union type for reference fields that looks like `Entry< … > | { sys: Link<"Entry"> }`. To access the entry you first have to check whether the entry is there.

TL;DR: Not a bug but uncertainty in types because we only have the full information at runtime.

**Comment by gigaSproule** [comment]
Ah, I was unaware that it wouldn't throw an error or something if the entries were not able to be fetched. Is this primarily for the case of broken links, or is it that the client won't return the entity if there are other issues such as network blips?

**Comment by veu** [comment]
It’s for broken links as there is no guarantee that the linked entry exists.

**Comment by gigaSproule** [comment]
Thanks for the correction.
